url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/com/wroblicky/andrew/joust/Util.java

```diff
@@ -0,0 +1,111 @@
+package com.wroblicky.andrew.joust;
+
+import java.util.Set;
+
+import com.wroblicky.andrew.joust.game.board.Location;
+import com.wroblicky.andrew.joust.game.chesspiece.ChessPiece;
+
+public final class Util {
+
+	public static String[][] getStringBoard(Set<ChessPiece> actives) {
+		String[][] newBoard = new String[8][8];
+		for (int i = 0; i < 8; i++) {
+			for (int j = 0; j < 8; j++) {
+				newBoard[i][j] = "-"; // denotes unoccupied
+			}
+		}
+		for (ChessPiece active : actives) {
+			int currX = active.getLocation().getXCoordinate();
+			int currY = active.getLocation().getYCoordinate();
+			newBoard[currX - 1][currY - 1] = active
+					.getChessPieceAllegianceType().toString();
+		}
+		return newBoard;
+	}
+
+	public static int reverseNum(int num) {
+		int toReturn = num;
+		if (num == 0) {
+			toReturn = 7;
+		} else if (num == 1) {
+			toReturn = 6;
+		} else if (num == 2) {
+			toReturn = 5;
+		} else if (num == 3) {
+			toReturn = 4;
+		} else if (num == 4) {
+			toReturn = 3;
+		} else if (num == 5) {
+			toReturn = 2;
+		} else if (num == 6) {
+			toReturn = 1;
+		} else if (num == 7) {
+			toReturn = 0;
+		}
+		return toReturn;
+	}
+
+	public static int fileToNum(String file) {
+		int toReturn = 0;
+		if (file.equals("a")) {
+			toReturn = 1;
+		} else if (file.equals("b")) {
+			toReturn = 2;
+		} else if (file.equals("c")) {
+			toReturn = 3;
+		} else if (file.equals("d")) {
+			toReturn = 4;
+		} else if (file.equals("e")) {
+			toReturn = 5;
+		} else if (file.equals("f")) {
+			toReturn = 6;
+		} else if (file.equals("g")) {
+			toReturn = 7;
+		} else if (file.equals("h")) {
+			toReturn = 8;
+		}
+		return toReturn;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 49:67, 71:89.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.